src/encord_active/lib/common/project.py

```diff
@@ -0,0 +1,214 @@
+from __future__ import annotations
+
+import dataclasses
+import itertools
+import json
+import logging
+from functools import partial
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+
+import encord.exceptions
+from encord import Project as EncordProject
+from encord.objects.ontology_structure import OntologyStructure
+from encord.orm.label_row import LabelRow
+from encord.project import LabelRowMetadata
+
+from encord_active.lib.common.utils import (
+    collect_async,
+    download_file,
+    fetch_project_meta,
+    slice_video_into_frames,
+)
+
+logger = logging.getLogger(__name__)
+encord_logger = logging.getLogger("encord")
+encord_logger.setLevel(logging.ERROR)
+
+
+@dataclasses.dataclass
+class Project:
+    image_paths: Dict[str, List[Path]]
+    label_rows: Dict[str, LabelRow]
+    label_row_meta: Dict[str, LabelRowMetadata]
+    ontology: OntologyStructure
+    project_hash: str
+    project_meta: Dict[str, str]
+
+    @staticmethod
+    def read(
+        cache_dir: Path,
+        subset_size: Optional[int] = None,
+        encord_project: Optional[EncordProject] = None,
+        **kwargs,
+    ) -> Project:
+        """
+        Fetches data either from cache or from the Encord project if it's available.
+
+        :param cache_dir: The root directory of the project.
+        :param subset_size: The number of label rows to fetch. None means all.
+        :param encord_project: An encord project for fetching data. If not provided, will only use cached files.
```

Comment:
[curious]🤔 

just to make sure I understand this, is the intended usage something like

```python
# setup project
project = Project.read("some/path", encord_project=EncordProject("whatever"))
project.save()

# in some other file, just use local project
project = Project.read("some/path")
```